### DISCUSSION TITLE
Error returned in executing first code example on "How to use promises" guide

### DISCUSSION THREAD

**Discussion by authentictech** [mdn/mdn-community#478] [Content] [answered]
I am getting an error in [this page](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises) in the first code example when trying to fetch using the browser console as directed.

The error is:

`Content-Security-Policy: The page's settings blocked the loading of a resource at https://mdn.github.io/learning-area/javascript/apis/fetching-data/can-store/products.json ("connect-src").`

The final response of the fetch command is:

`Promise { <state>: "rejected", <reason>: TypeError }
​
<state>: "rejected"
​
<reason>: TypeError: NetworkError when attempting to fetch resource.
​
<prototype>: Promise.prototype { … }
debugger eval code:5:9
`

The error does not occur when adding the code to a web page.

Does anyone else get this error?

**Comment by bsmth** [comment, marked as answer]
This is because `mdn.github.io` is not set in [MDN's current CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP). You can give this a try in your console from any MDN page:

```js
const fetchPromise = fetch(
  "https://developer.mozilla.org/en-US/docs/Learn/contributors.txt",
);

console.log(fetchPromise);

fetchPromise.then((response) => {
  console.log(`Received response: ${response.status}`);
});

console.log("Started request…");
```

Alternatively, if you wanted to try the example in the page, specifically, you could put it in a HTML document and open it in your browser:

```html
<!DOCTYPE html>
<title>Run fetch</title>
<script>
  const fetchPromise = fetch(
    "https://mdn.github.io/learning-area/javascript/apis/fetching-data/can-store/products.json"
  );

  console.log(fetchPromise);

  fetchPromise.then((response) => {
    console.log(`Received response: ${response.status}`);
  });

  console.log("Started request…");
</script>
```

```log
14:40:02.724 Promise { <state>: "pending" } [cspjs.html:8:11](file:///path/to/cspjs.html)
14:40:02.724 Started request… [cspjs.html:14:11](file:///path/to/cspjs.html)
14:40:02.782 Received response: 200 [cspjs.html:11:13](file:///path/to/cspjs.html)
```

I think we may open a content issue to replace the URL in those examples (or check if `mdn.github.io` needs to be added to CSP).

**Reply by authentictech** [reply]
Thank you. I wanted to be sure whether or not something did need to be updated in the docs before making any edits. However, I think I will leave the editing for others to do as I am not aware of which pages are suitable to use instead.

**Reply by bsmth** [reply]
You're welcome! I also just noticed that Will has pointed out the [docs suggest visiting `example.com`](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises#using_the_fetch_api) and executing the examples in the console from there.

**Reply by wbamberg** [reply]
> I think we may open a content issue to replace the URL in those examples (or check if mdn.github.io needs to be added to CSP).

We could do one of those things (more likely the first I would think), but there would still need to be explicit instructions about how to run the examples. For instance, if someone opens a new blank tab to run the examples in (which might be a fairly natural thing to try) then the examples still won't work. So either way, we need explicit instructions, and people have to read them.
